src/puppetlabs/puppetdb/query_eng/engine.clj

```diff
@@ -1526,6 +1526,19 @@
   [x]
   (instance? clojure.lang.IPersistentVector x))
 
+(defn remove-elided-nodes
+  "This step removes elided nodes (marked with ::elide by expand-user-query) from
+  `and` and `or` clauses."
+  [node]
+  (cm/match [node]
+            [["and" & clauses]]
+            (into ["and"] (remove #(= ::elide %) clauses))
+
+            [["or" & clauses]]
+            (into ["or"] (remove #(= ::elide %) clauses))
```

Comment:
Is there any way this can happen outside of an "and" or "or"?

Reply:
what other case are you thinking of?

Reply:
maybe if you just had node_state== any by itself? it seems like that would get ::elide, but not hit iether of these cases.